### PR TITLE
ROX-14221: upgrade procedures for cloud

### PR DESCRIPTION
**Pull request by kcarmichael08**
Version(s):
4.4+

[Issue](https://issues.redhat.com/browse/ROX-14221)

Links to docs previews:

https://78259--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/upgrading-cloud/upgrade-cloudsvc-operator
https://78259--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/upgrading-cloud/upgrade-cloudsvc-helm
https://78259--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl

QE review:
- [x] QE has approved this change. (**ACS has no QE, reviewed/ack'd by ACS SME**)

Additional information:

Note to technical reviewers:
This material is reused from the RHACS docs where applicable. I have removed references to Central, but if there are things in these docs that do not apply to Cloud, please mark in your review. Thank you!

### CODE REVIEW

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+:context: upgrade-cloudsvc-roxctl
+
+toc::[]
+
+[role="_abstract"]
+You can upgrade your secured clusters in {product-title-managed-short} by using the `roxctl` CLI.
+
+[IMPORTANT]
+====
+You need to perform the manual upgrade procedure only if you used the `roxctl` CLI to install {product-title-managed-short}.
```
Comment (vladbologa) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc:14]: This renders as
>  You need to perform the manual upgrade procedure only if you used the roxctl CLI to install RHACS Cloud Service.

which I think should be something like `to install the secured cluster(s)`

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+You are responsible for regularly upgrading the version of {product-title-short} on your secured clusters to ensure compatibility with {product-title-managed-short}.
+
+include::modules/prepare-operator-upgrades.adoc[leveloffset=+1]
+include::modules/change-collection-method.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+.Additional resources
+* link:https://docs.openshift.com/container-platform/latest/operators/admin/olm-upgrading-operators.html[Updating installed Operators]
+
+//does this apply to Cloud Service? Or only updating Central components?
```
Comment (vladbologa) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc:22]: Good question. I doubt that there are ACS CS customers that have secured clusters on the 3.7x versions (possibly some remaining from the "service preview" in late 2022), so we could skip this part about changing channels.

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+include::modules/operator-upgrade-change-subscription-channel.adoc[leveloffset=+1]
+
+[id="rollback-operator-upgrade-cloud"]
+== Rolling back an Operator upgrade for secured clusters
```
Comment (vladbologa) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc:26]: I wonder if we should have this section.

We try to not roll back Central Services in ACS CS. We have canary deployments and we wouldn't deploy a new version to customers if it's problematic. Which should mean that customers shouldn't have to roll back their secured clusters either.

OTOH, there could be the possibility that there's a problem with the secured cluster specifically and has to be rolled back, so we could also keep this section.

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+:context: upgrade-cloudsvc-helm
+
+toc::[]
+
+[role="_abstract"]
+You can upgrade your secured clusters in {product-title-managed-short} by using Helm charts.
+
+If you have installed {product-title-short} secured clusters by using Helm charts, to upgrade to the latest version of {product-title-short} perform the following steps:
+
+. Update the Helm chart.
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc:14]:
```suggestion

```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+. Run the `helm upgrade` command.
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc:15]:
```suggestion

```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+[role="_additional-resources"]
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc:17]:
```suggestion

```
In the preview, it looks like maybe the "Updating the Helm chart repository" module is mixed up with an "Additional resources" section.

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+.Additional resources
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-helm.adoc:18]:
```suggestion

```

File: modules/upgrade-helm-chart.adoc [hunk 8]
```diff
@@ -23,10 +30,22 @@ $ helm upgrade -n stackrox stackrox-central-services \
   --set central.db.serviceTLS.generate=true \
   --set central.db.persistence.persistentVolumeClaim.createClaim=true
 ----
+endif::cloud-svc[]
+
 +
+[source,terminal]
+----
+$ helm upgrade -n stackrox stackrox-secured-cluster-services \
+  rhacs/secured-cluster-services --version <current-rhacs-version> \// <1>
```
Comment (lahinson) [inline modules/upgrade-helm-chart.adoc:39]: I don't see a corresponding callout for the <1>.

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+include::modules/common-attributes.adoc[]
+:context: upgrade-cloudsvc-operator
+
+toc::[]
+
+[role="_abstract"]
+
+Red{nbsp}Hat provides regular service updates for the components that it manages, including Central services. These service updates include upgrades to new versions of {product-title-managed}.
+
+You are responsible for regularly upgrading the version of {product-title-short} on your secured clusters to ensure compatibility with {product-title-managed-short}.
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc:13]:
```suggestion
You must regularly upgrade the version of {product-title-short} on your secured clusters to ensure compatibility with {product-title-managed-short}.
```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+include::modules/change-collection-method.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+.Additional resources
+* link:https://docs.openshift.com/container-platform/latest/operators/admin/olm-upgrading-operators.html[Updating installed Operators]
+
+[id="rollback-operator-upgrade-cloud"]
+== Rolling back an Operator upgrade for secured clusters
+
+To roll back an Operator upgrade, you must perform the steps described in one of the following sections. You can roll back an Operator upgrade by using the CLI or the {ocp} web console.
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc:25]:
```suggestion
To roll back an Operator upgrade, you can use either the CLI or the {ocp} web console.
```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+[NOTE]
+====
+Rolling back Operator upgrades on secured clusters is needed only in rare cases, for example, if there is an issue with the secured cluster.
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-operator.adoc:29]:
```suggestion
On secured clusters, rolling back Operator upgrades is needed only in rare cases, for example, if an issue exists with the secured cluster.
```

File: modules/prepare-operator-upgrades.adoc [hunk 12]
```diff
@@ -1,13 +1,24 @@
...
 :_mod-docs-content-type: CONCEPT
 [id="prepare-operator-upgrades_{context}"]
 = Preparing to upgrade
 
+ifeval::["{context}" == "upgrade-cloudsvc-operator"]
+:cloud-svc:
+endif::[]
+
 [role="_abstract"]
 Before you upgrade the {rh-rhacs-first} version, you must perform the following steps:
```
Comment (lahinson) [inline modules/prepare-operator-upgrades.adoc:14]:
```suggestion
Before you upgrade the {rh-rhacs-first} version, complete the following steps:
```

File: modules/prepare-operator-upgrades.adoc [hunk 13]
```diff
@@ -1,13 +1,24 @@
...
 
+ifndef::cloud-svc[]
 * If you are upgrading from version 3.74, verify that you are running the latest patch release version of the {product-title-short} Operator 3.74.
 * Backup your existing Central database.
+endif::cloud-svc[]
 * If the cluster you are upgrading contains the `SecuredCluster` custom resource (CR), change the collection method to CORE_BPF. For more information, see "Changing the collection method".
```
Comment (lahinson) [inline modules/prepare-operator-upgrades.adoc:20]:
```suggestion
* If the cluster you are upgrading contains the `SecuredCluster` custom resource (CR), change the collection method to `CORE_BPF`. For more information, see "Changing the collection method".
```

File: modules/rollback-operator-upgrades-console.adoc [hunk 14]
```diff
@@ -16,6 +21,7 @@ You can roll back the Operator version by using the {ocp} web console.
 . Go to the *Operators* -> *Installed Operators* page.
 . Locate the {product-title-short} Operator and click on it.
```
Comment (lahinson) [inline modules/rollback-operator-upgrades-console.adoc:22]:
```suggestion
. Click the {product-title-short} Operator.
```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc [hunk 15]
```diff
@@ -0,0 +1,68 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="upgrade-cloudsvc-roxctl"]
+= Manually upgrading secured clusters in RHACS Cloud Service by using the `roxctl` CLI
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc:3]: From the [OCP documentation guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc): "Do not use backticks or other markup in assembly or module headings."
```suggestion
= Manually upgrading secured clusters in RHACS Cloud Service by using the roxctl CLI
```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+:context: upgrade-cloudsvc-roxctl
+
+toc::[]
+
+[role="_abstract"]
+You can upgrade your secured clusters in {product-title-managed-short} by using the `roxctl` CLI.
+
+[IMPORTANT]
+====
+You need to perform the manual upgrade procedure only if you used the `roxctl` CLI to install the secured clusters.
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc:14]:
```suggestion
You need to manually upgrade secured clusters only if you used the `roxctl` CLI to install the secured clusters.
```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+====
+
+[id="upgrade-roxctl-cli"]
+== Upgrading the `roxctl` CLI
+
+To upgrade the `roxctl` CLI to the latest version you must uninstall the existing version of `roxctl` CLI and then install the latest version of the `roxctl` CLI.
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc:20]:
```suggestion
To upgrade the `roxctl` CLI to the latest version, you must uninstall your current version of `roxctl` CLI and then install the latest version of the `roxctl` CLI.
```

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+include::modules/uninstall-roxctl-cli.adoc[leveloffset=+2]
+include::modules/install-roxctl-cli-linux.adoc[leveloffset=+2]
+include::modules/install-roxctl-cli-macos.adoc[leveloffset=+2]
+include::modules/install-roxctl-cli-windows.adoc[leveloffset=+2]
+
+[id="upgrade-secured-clusters-cloud"]
+== Upgrading all secured clusters manually
+
+[IMPORTANT]
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc:30]: I'm not sure why this admonition is separated into nested lists. In the preview, it makes the red box take up a lot of the page. Consider revising the wording to something like this to make the page easier to scan:
`* If you are using automatic upgrades, update all your secured clusters by using automatic upgrades.`
`* If you are not using automatic upgrades, complete the instructions in this section on all secured clusters. To ensure optimal functionality, use the same {product-title-short} version for your secured clusters that {product-title-managed-short} is running.`

File: cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+
+To complete manual upgrades of each secured cluster running Sensor, Collector, and Admission controller, follow these instructions.
+
+include::modules/update-other-images.adoc[leveloffset=+2]
+
+.Next steps
+* xref:../../cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc#verify-secured-cluster-upgrade_upgrade-cloudsvc-roxctl[Verify the secured cluster upgrade]
+
+
+[Verifying secured cluster upgrade]
```
Comment (lahinson) [inline cloud_service/upgrading-cloud/upgrade-cloudsvc-roxctl.adoc:46]: Should line 46 be removed?

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 20]
```diff
@@ -42,7 +51,8 @@ Name:      sensor-84545f86b7-xgdwf
 +
 In this example, you can see that each pod has its own custom SCC, which is specified through the `openshift.io/scc` field.
 
-. Add the required roles and role bindings to use the {osp} SCCs instead of the {product-title-short} custom SCCs. 
+. Add the required roles and role bindings to use the {osp} SCCs instead of the {product-title-short} custom SCCs.
+ifndef::cloud-svc[]
 +
 To add the required roles and role bindings to use the {osp} SCCs for the Central cluster, perform the following steps:
```
Comment (lahinson) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:57]:
```suggestion
To add the required roles and role bindings to use the {osp} SCCs for the Central cluster, complete the following steps:
```

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 21]
```diff
@@ -212,7 +222,7 @@ subjects:
 ----
 $ oc -n stackrox create -f ./update-central.yaml
 ----
-
+endif::cloud-svc[]
 . To add the required roles and role bindings to use the {osp} SCCs for all secured clusters, perform the following steps:
```
Comment (lahinson) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:226]:
```suggestion
. To add the required roles and role bindings to use the {osp} SCCs for all secured clusters, complete the following steps:
```

File: modules/operator-upgrade-change-subscription-channel.adoc [hunk 22]
```diff
@@ -1,21 +1,33 @@
...
+endif::[]
 
 [role="_abstract"]
 You can change the update channel for the {product-title-short} Operator by using the {ocp} web console or by using the command line. For upgrading to {product-title-short} 4.0 from {product-title-short} 3.74, you must change the update channel.
 
 [IMPORTANT]
 ====
-You must change the subscription channel for all clusters where you have installed {product-title-short} Operator, including Central and all Secured clusters.
+ifndef::cloud-svc[]
+You must change the subscription channel for all clusters where you have installed the {product-title-short} Operator, including Central and all Secured clusters.
```
Comment (lahinson) [inline modules/operator-upgrade-change-subscription-channel.adoc:19]:
```suggestion
You must change the subscription channel for all clusters where you installed the {product-title-short} Operator, including Central and all Secured clusters.
```

**Review by lahinson — COMMENTED** [review]
Nice job on this large doc set! I made a few suggestions based on IBM Style and the OCP doc guidelines.

File: modules/operator-upgrade-change-subscription-channel.adoc [hunk 23]
```diff
@@ -47,4 +59,8 @@ $ oc -n rhacs-operator \ <1>
 ----
 <1> If you use Kubernetes, enter `kubectl` instead of `oc`.
 
-During the update the {product-title-short} Operator provisions a new deployment called `central-db` and your data begins migrating. It takes around 30 minutes and only happens once when you upgrade.
+During the update the {product-title-short} Operator provisions a new deployment called `central-db` and your data begins migrating. It takes around 30 minutes and only happens after when you upgrade.
```
[the remark this answers is not in the document]
Comment (kcarmichael08) [reply modules/operator-upgrade-change-subscription-channel.adoc:62]: This is existing text, so I will ask (I don't know the answer).

**Pull request merged**